analyses/cnv-chrom-plot/cn_status_heatmap.Rmd

```diff
@@ -0,0 +1,452 @@
+
+### Usage
+
+This notebook can be run via the command line from the top directory of the 
+repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/cnv-chrom-plot/cn_status_heatmap.Rmd', 
+                              clean = TRUE)"
+```
+
+### Cutoffs: 
+
+```{r}
+# The max length of a segment to use the data.
+# segments that are too long may dominate the heatmap and/or be indicators of
+# broader structural changes
+length_max <- 1e7
+
+# Set minimum percentage of a bin that should be callable to report data.
+frac_uncallable <- 0.75
+
+# Absolute fraction needed for a bin to be called a particular status
+frac_threshold <- 0.75
+```
+
+### Set Up
+
+```{r}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+### Directories and Files
+
+```{r}
+# Path to input directory
+input_dir <- file.path("..", "..", "data")
+figure_dir <- file.path("..", "..", "figures")
+scratch_dir <- file.path("..", "..", "scratch")
+
+# Path to output directory
+plots_dir <- "plots"
+
+# Create the plots_dir if it does not exist
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+Read in custom functions.
+
+```{r}
+source(file.path("util", "bin-coverage.R"))
+```
+
+Import color palettes. 
+
+```{r}
+# Import standard color palettes for project
+histology_col_palette <- readr::read_tsv(
+  file.path(figure_dir, "palettes", "histology_color_palette.tsv")
+) %>%
+  # We'll use deframe so we can use it as a recoding list
+  tibble::deframe()
+```
+
+Read in the divergent color palette and set it up with three colors. 
+In this instance, we only need three values for `gain`, `neutral`, and `loss`.
+
+```{r}
+divergent_col_palette <- readr::read_tsv(
+  file.path(figure_dir, "palettes", "divergent_color_palette.tsv")
+) %>%
+  # Only keep only these three colors
+  dplyr::filter(
+    color_names %in% c("divergent_low_4", "divergent_neutral", "divergent_high_4")
+  ) %>%
+  dplyr::pull("hex_codes")
+```
+
+### Read in data 
+
+```{r}
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(input_dir, "pbta-histologies.tsv")) %>%
+  # Easier to deal with NA short histologies if they are labeled something different
+  dplyr::mutate(short_histology = as.character(tidyr::replace_na(short_histology, "none"))) %>%
+  # Tack on the sample color using the short_histology column and a recode
+  dplyr::mutate(sample_color = dplyr::recode(
+    short_histology,
+    !!!histology_col_palette
+  ))
+```
+
+### Set up consensus copy number data
+
+```{r}
+# Read in the segment copy number data
+seg_data <- data.table::fread(file.path(
+  input_dir,
+  "pbta-cnv-consensus.seg.gz"
+),
+data.table = FALSE
+)
+```
+
+Set up the status for each consensus segment. 
+
+```{r}
+seg_data <- seg_data %>%
+  # Join the histology column to this data
+  dplyr::inner_join(dplyr::select(
+    metadata,
+    "Kids_First_Biospecimen_ID",
+    "short_histology",
+    "tumor_ploidy"
+  ),
+  by = c("ID" = "Kids_First_Biospecimen_ID")
```

Comment:
Same thing... try not to open 2 sets of parens on the same line that don't then close together, as it makes for non-semantic indentation.
```suggestion
seg_data <- seg_data %>%
  # Join the histology column to this data
  dplyr::inner_join(
    dplyr::select(
      metadata,
      "Kids_First_Biospecimen_ID",
      "short_histology",
      "tumor_ploidy"
    ),
    by = c("ID" = "Kids_First_Biospecimen_ID")
```